ibm/data_source_ibm_is_instance_group_memberships.go

```diff
@@ -0,0 +1,199 @@
+										Computed:    true,
+										Description: "The unique identifier for this instance template.",
+									},
+									"name": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The unique user-defined name for this instance template.",
+									},
+								},
+							},
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The user-defined name for this instance group membership. Names must be unique within the instance group.",
+						},
+						"load_balancer_pool_member": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The unique identifier for this load balancer pool member.",
+						},
+						"status": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The status of the instance group membership- `deleting`: Membership is deleting dependent resources- `failed`: Membership was unable to maintain dependent resources- `healthy`: Membership is active and serving in the group- `pending`: Membership is waiting for dependent resources- `unhealthy`: Membership has unhealthy dependent resources.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMISInstanceGroupMembershipsRead(d *schema.ResourceData, meta interface{}) error {
+	log.Println("Inside dataSourceIbmIsInstanceGroupMembershipsRead")
+
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	instanceGroupID := d.Get("instance_group").(string)
+	log.Println("instanceGroupID")
+	log.Println(instanceGroupID)
+
+	// Support for pagination
+	start := ""
+	allrecs := []vpcv1.InstanceGroupMembership{}
+
+	for {
+		listInstanceGroupMembershipsOptions := vpcv1.ListInstanceGroupMembershipsOptions{
+			InstanceGroupID: &instanceGroupID,
+		}
+		log.Println(instanceGroupID)
+		instanceGroupMembershipCollection, response, err := sess.ListInstanceGroupMemberships(&listInstanceGroupMembershipsOptions)
+		if err != nil {
+			return fmt.Errorf("Error Getting InstanceGroup Membership Collection %s\n%s", err, response)
+		}
+
+		start = GetNext(instanceGroupMembershipCollection.Next)
+		allrecs = append(allrecs, instanceGroupMembershipCollection.Memberships...)
+
+		if start == "" {
+			break
+		}
+
+	}
+
+	memberships := make([]map[string]interface{}, 0)
+	for _, instanceGroupMembership := range allrecs {
+
+		membership := map[string]interface{}{
+			"delete_instance_on_membership_delete": *instanceGroupMembership.DeleteInstanceOnMembershipDelete,
+			"instance_group_membership":            *instanceGroupMembership.ID,
+			"name":                                 *instanceGroupMembership.Name,
+			"status":                               *instanceGroupMembership.Status,
+		}
+
+		instances := make([]map[string]interface{}, 0)
+		if instanceGroupMembership.Instance != nil {
+			log.Println("*instanceGroupMembership.Instance.CRN - Instance")
```

Comment:
Remove log stmts which are not required

Reply:
@hkantare Removed Logs, Can you please review and merge the PR..